lib/bank_account.ex

```diff
@@ -8,31 +10,78 @@ defmodule BankAccount do
   """
   @opaque account :: pid
 
+  @impl true
+  def init(:ok) do
+    {:ok, %{balance: 0, status: :account_open}}
+  end
+
+  def start_link do
+    GenServer.start_link(__MODULE__, :ok)
+  end
+
+  @impl true
+  # balance inquiry when status: :account_closed
+  def handle_call(:balance, _from, %{status: :account_closed} = state) do
+    {:reply, {:error, :account_closed}, state}
+  end
+
+  @impl true
+  # balance inquiry
+  def handle_call(:balance, _from, state) do
+    {:reply, Map.get(state, :balance), state}
+  end
```

Comment:
```suggestion
  def handle_call(:balance, _from, %{balance: balance}) do
    {:reply, balance, state}
  end
```

In some of the calls, you are pattern matching in the function head to access state values, and others you are using `Map.get`. I really prefer pattern matching when possible, as I find it clearer to read. Either way it would be good to unify on one approach.